### DISCUSSION TITLE
Asset Metadata

### DISCUSSION THREAD

**Discussion by nkapetanas** [eclipse-edc/Connector#4242] [Q&A]
Hi all! 

I would like to as some things about the asset metadata. When we create an asset we send to the connector a request with a json-ld like: 
```
{
  "@context": {
    "@vocab": "https://w3id.org/edc/v0.0.1/ns/"
  },
  "@id": "assetId",
  "properties": {
    "name": "product description",
    "contenttype": "application/json"
  },
  "dataAddress": {
    "type": "HttpData",
    "name": "Test asset",
    "baseUrl": "https://jsonplaceholder.typicode.com/users",
    "proxyPath": "true"
  }
}
```

1. The first question is, in the "properties" field can I add as many properties as I want and in the way I want, like "key": "value" pairs? 

2. My second question is, can these metadata have properties that are based on some rdf or xml shema? And if yes how we declare it in order the connector to understand that some metadata properties are an rdf or an xml schema? is there any example?

3. If we have a Connector that we want to specifically accept assets with metadata properties that are exact, makes sense to create an extention that before the actuall creation of the asset, we validate the all the metadata provided are existing?

Thank you all for your time :D :D

**Comment by jimmarino** [comment]
Some responses:

1. Use a Json-Ld context for this, including the prefix in the key
2. The connector doesn't understand RDF or other types of metadata "decorators." If you have a Json-Ld context, you can define types that are then defined using a schema language.
3. You can write a validator extension that operates on the expanded Json-Ld (see the code for examples of how to do this) and performs checks against schema-defined constraints.

**Reply by nkapetanas** [reply]
Thank you for your answer :) Can you provide a dummy example for (1) and (2) ? For example if I want my asset to have as metadata "column1": "temperature" and "column2": "air speed", how it should be formatted?

**Reply by jimmarino** [reply]
There are plenty of examples of how to use the management API and how to write Json-Ld. @paullatzelsperger also pointed you to the Json-Ld playground.

**Reply by nkapetanas** [reply]
Alright, thank you!

**Comment by paullatzelsperger** [comment]
hey @nkapetanas. 
adding to Jim's response, you could check out the [JSON-LD Playground](https://json-ld.org/playground/). That'll give you a low-friction way to try out expanding your assets.

also please format your code with markdown code fences (```). makes it much easier to read.

**Reply by nkapetanas** [reply]
Thank you!
